src/NServiceBus.Core/Recoverability/Faults/MoveFaultsToErrorQueueNoTransactionBehavior.cs

```diff
@@ -0,0 +1,72 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using Logging;
+    using Pipeline;
+    using Transports;
+
+    // TODO: Remove duplication in MoveFaultsToErrorQueueBehavior and MoveFaultsToErrorQueueNoTransactionBehavior.
+    class MoveFaultsToErrorQueueNoTransactionBehavior : ForkConnector<ITransportReceiveContext, IFaultContext>
+    {
+        public MoveFaultsToErrorQueueNoTransactionBehavior(CriticalError criticalError, string errorQueueAddress, string localAddress)
+        {
+            this.criticalError = criticalError;
+            this.errorQueueAddress = errorQueueAddress;
+            this.localAddress = localAddress;
+        }
+
+        public override async Task Invoke(ITransportReceiveContext context, Func<Task> next, Func<IFaultContext, Task> fork)
+        {
+            try
+            {
+                await next().ConfigureAwait(false);
+            }
+            catch (Exception exception)
+            {
+                try
```

Comment:
If we extracted `MoveToErrorQueueMethod` maybe we could merge transactional and nontransactional behavior and just pass `TransportTransactionMode` as one of the parameters.